src/RemoteRequest/TemporaryFileCachedRemoteGetRequest.php

```diff
@@ -0,0 +1,209 @@
+    {
+        try {
+            $response = $this->remoteGetRequest->get($url, $headers);
+            $this->cacheResponse($url, $response);
+            return $response;
+        } catch (Exception $error) {
+            throw FailedToGetCachedResponse::withUrl($url);
+        }
+    }
+
+    /**
+     * Save the response in temporary file.
+     *
+     * @param string   $url      The request url.
+     * @param Response $response The response that need to be cached.
+     */
+    private function cacheResponse($url, Response $response)
+    {
+        file_put_contents($this->getTemporaryFilePath($url), serialize($response));
+    }
+
+    /**
+     * Check whether the request is expired.
+     *
+     * @param string                   $file     The absolute path of the file that contains the response data.
+     * @param RemoteGetRequestResponse $response The response that need to be checked.
+     * @return bool
+     */
+    private function isExpired($file, RemoteGetRequestResponse $response)
+    {
+        $expiry = $this->getExpiryTime($file, $response);
+
+        return microtime(true) > $expiry->getTimestamp();
+    }
+
+    /**
+     * Calculate the expiry time.
+     *
+     * @param string                   $file     The absolute path of the file that contains the response data.
+     * @param RemoteGetRequestResponse $response The response data need to be calculated.
+     *
+     * @return DateTimeImmutable Cache expiry time object.
+     */
+    private function getExpiryTime($file, RemoteGetRequestResponse $response)
+    {
+        $expiry           = self::EXPIRY_TIME;
+        $fileModifiedTime = (new DateTimeImmutable())->setTimestamp(filemtime($file));
+
+        if ($response->hasHeader(self::CACHE_CONTROL)) {
+            $maxAge = $this->getMaxAge($response->getHeader(self::CACHE_CONTROL));
+            $expiry = ($maxAge >= 0) ? $maxAge : $expiry;
+        }
+
+        return $fileModifiedTime->modify("{$expiry} sec");
+    }
+
+    /**
+     * Extract the max-age from response header.
+     *
+     * @param array $cache_control_strings Cache-Control header value.
+     * @return int The max-age value of the Cache-Control header.
+     */
+    private function getMaxAge($cache_control_strings)
+    {
+        $max_age = -1;
+
+        foreach ((array) $cache_control_strings as $cache_control_string) {
+            $cache_control_parts = array_map('trim', explode(',', $cache_control_string));
+
+            foreach ($cache_control_parts as $cache_control_part) {
+                $cache_control_setting_parts = array_map('trim', explode('=', $cache_control_part));
+
+                if (count($cache_control_setting_parts) !== 2) {
+                    continue;
+                }
+
+                if ('max-age' === $cache_control_setting_parts[0]) {
+                    $max_age = abs((int)$cache_control_setting_parts[1]);
+                }
+            }
+        }
+
+        return $max_age;
+    }
```

Comment:
We're using camelCasing for variable names.
```suggestion
     * @param array $cacheControlStrings Cache-Control header value.
     * @return int The max-age value of the Cache-Control header.
     */
    private function getMaxAge($cacheControlStrings)
    {
        $maxAge = -1;

        foreach ((array) $cacheControlStrings as $cacheControlString) {
            $cacheControlParts = array_map('trim', explode(',', $cacheControlString));

            foreach ($cacheControlParts as $cacheControlPart) {
                $cacheControlSettingParts = array_map('trim', explode('=', $cacheControlPart));

                if (count($cacheControlSettingParts) !== 2) {
                    continue;
                }

                if ('max-age' === $cacheControlSettingParts[0]) {
                    $maxAge = abs((int)$cacheControlSettingParts[1]);
                }
            }
        }

        return $maxAge;
    }
```